### ISSUE TITLE
linter: `eslint/no-unneeded-ternary` with `defaultAssignment: 'false'` flags ternaries with type assertions

### ISSUE THREAD

**Issue by jeanne-cai** [oxc-project/oxc#22765]
### What version of Oxlint are you using?

1.67.0

### What command did you run?

`yarn oxlint -c oxlint.config.mjs`

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```ts
import { defineConfig } from 'oxlint';

export default defineConfig({
    categories: {
        correctness: 'off',
    },
    rules: {
        'eslint/no-unneeded-ternary': [
            'error',
            {
                defaultAssignment: false,
            },
        ],
    },
});
```

### What happened?

The rule incorrectly treats ternaries with type assertions as unnecessary.
This is no longer the plain `x ? x : y` pattern that `defaultAssignment: false` is meant to target.

To reproduce:
```ts
// correctly flagged
const a = foo ? foo : undefined;

// incorrectly flagged
const b = foo ? (foo as string) : undefined;
```

Output
```bash
  × eslint(no-unneeded-ternary): Unnecessary use of conditional expression for default assignment
   ╭─[test.ts:2:11]
 1 │ // correctly flagged
 2 │ const a = foo ? foo : undefined;
   ·           ─────────────────────
 3 │ 
   ╰────
  help: Remove this ternary operator and use the variable directly

  × eslint(no-unneeded-ternary): Unnecessary use of conditional expression for default assignment
   ╭─[test.ts:5:11]
 4 │ // incorrectly flagged
 5 │ const b = foo ? (foo as string) : undefined;
   ·           ─────────────────────────────────
   ╰────
  help: Remove this ternary operator and use the variable directly
```

**Comment by camc314** [comment]
Hi! thanks for reporting this.

I think Oxlint's behaviour is reasonable here.

The spirit of `no-unneeded-ternary` is to catch conditional expressions that can be expressed more directly. In this case:
```
foo ? (foo as string) : undefined
```
is still the same default-assignment pattern as
```
foo ? foo : undefined
```
the type assertion does not change the runtime value of the consequent.

The original code snippet can be expressed as:
```
foo as string || undefined
```

Eslint currently does not report this because the core rule only knows about JS syntax, and has no knowledge of the typescript `TSAsExpression`.


Let me know if you think this is reasonable - thanks!

**Comment by jeanne-cai** [comment]
Hi! Thanks for the clarification, that makes sense. I hadn't thought about it that way.
